src/api/authorize.ts

```diff
@@ -67,8 +67,8 @@ export default function authorize(
     response_type: 'code',
     scope: constructScopes(scopes),
     redirect_uri: redirectUri,
-    code_challenge: codeChallenge,
-    code_challenge_method: codeVerifier && 'S256',
+    code_challenge: codeVerifier && codeChallenge,
```

Comment:
```suggestion
    code_challenge: codeChallenge || undefined,
```
To avoid checking `codeVerifier` twice.